src/main/java/com/google/api/codegen/metacode/InitValueConfig.java

```diff
@@ -74,6 +76,15 @@ public InitValueConfig withInitialCollectionValue(String entityName, InitValue v
     return toBuilder().setResourceNameBindingValues(ImmutableMap.of(entityName, value)).build();
   }
 
+  public InitValueConfig withUpdatedInitialCollectionValue(String entityName, InitValue value) {
+    Map<String, InitValue> map = new HashMap<>();
```

Comment:
Do you need this intermediate map? Can you just do the two operations directly on the `ImmutableMap.<>builder` you create in line 84?

Reply:
Because `entityName` will be a duplicate key. That will cause the `build()` to [fail](https://google.github.io/guava/releases/21.0/api/docs/com/google/common/collect/ImmutableMap.Builder.html#put-K-V-).

Reply:
Do you have to pass an `ImmutableMap`? Creating a map just so it can be copied to another map seems less than elegant. Hopefully `putAll` is efficient and doesn't invove a lot of memory shuffling.